libs/autofile/autofile.go

```diff
@@ -58,18 +60,18 @@ func OpenAutoFile(path string) (af *AutoFile, err error) {
 
 func (af *AutoFile) Close() error {
 	af.ticker.Stop()
+	close(af.tickerStopped)
 	err := af.closeFile()
 	sighupWatchers.removeAutoFile(af)
 	return err
 }
 
 func (af *AutoFile) processTicks() {
-	for {
```

Comment:
was there a reason for the removal of the for loops? surprised we didn't catch it in review, but seems to be the source of https://github.com/tendermint/tendermint/issues/2259

Reply:
😱 I think these got removed accidentally.